### PR TITLE
Created Repository, Interface and methods for UserAssignedTest

### PR DESCRIPTION
**Pull request by AlpaslanPolat**
Created User Assigned Test

### CODE REVIEW

File: exalearn-api/ExaLearn.Bl/Services/UserService.cs [hunk 1]
```diff
@@ -44,5 +46,11 @@ public async Task<HrHistoryDTO[]> GetHrUserHistoryByIdAsync(int id)
             var passedTests = await _historyRepository.GetHRUserHistoryByIdAsync(id);
             return _mapper.Map<HrHistoryDTO[]>(passedTests);
         }
+
+        public async Task<UserAssignedTestDTO[]> GetUserAssignedTestByIdAsync(int id)
+        {
+            var assignedTest = await _assignedTestRepository.GetUserAssignedTestByIdAsync(id);
+            return _mapper.Map<UserAssignedTestDTO[]>(assignedTest);
+        }
     }
```
Comment (artemgael96) [inline exalearn-api/ExaLearn.Bl/Services/UserService.cs:55]: add test

File: exalearn-api/ExaLearn.Dal/Repositories/AssignTestRepository.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace ExaLearn.Dal.Repositories
+{
+    public class AssignTestRepository : GenericRepository<AssignTest>, IAssignTestRepository
+    {
+        public AssignTestRepository(ExaLearnDbContext appDbContext) : base(appDbContext)
+        {
+        }
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Dal/Repositories/AssignTestRepository.cs:16]: add space

File: exalearn-api/ExaLearn.WebApi/Controllers/UserController.cs [hunk 3]
```diff
@@ -40,5 +40,11 @@ public async Task<IActionResult> GetHrUserHistoryById(int id)
         {
             return Ok(await _userService.GetHrUserHistoryByIdAsync(id));
         }
+
+        [HttpGet("{id}/UserAssignedTest")]
```
Comment (romanstat) [inline exalearn-api/ExaLearn.WebApi/Controllers/UserController.cs]: userAssignedTest

File: exalearn-api/ExaLearn.Dal/Interfaces/IAssignTestRepository.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
+﻿using ExaLearn.Dal.Entities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Dal/Interfaces/IAssignTestRepository.cs]: check all files and remove useless usings

**Review by romanstat — APPROVED** [review]

**Comment by romanstat** [comment]
merge dev into your branch, resolve conflicts
image

**Pull request closed without merging**